hal/src/common/thumbv7em/sercom/v1/pad_map.rs

```diff
@@ -0,0 +1,366 @@
+//! Implementations of the [`pads`] [`Map`] trait
+//!
+//! This module provides implementations of [`Map`] that are part of a
+//! compatibility shim for the [`v1`] API. The [`v1`] API did not account for
+//! the concept of [`IoSet`] in SAMD5x & SAME5x chips. Instead, it mapped pins
+//! directly to SERCOM pads. It was possible to accidentally mix pins from
+//! different IOSETs, which would not work.
+//!
+//! This module provides implementations of [`Map`] directly on [`PinId`]s. The
+//! [`v1::pads`] shim will then transfer the mapping from [`PinId`] to
+//! configured pins.
+//!
+//! [`v1`]: crate::sercom::v1
+//! [`v1::pads`]: crate::sercom::v1::pads
+//! [`pads`]: crate::sercom::v2::pads
+//! [`IoSet`]: crate::sercom::v2::pads::IoSet
+//! [`Map`]: crate::sercom::v2::pads::Map
+//! [`PinId`]: crate::gpio::v2::PinId
+
+use crate::gpio::v2::*;
+use crate::sercom::v2::pads::*;
+
+//==============================================================================
+//  Pad definitions
+//==============================================================================
+
+// This macro accepts three different patterns to handle configuration
+// attributes applied to both individual PadNums as well as entire IoSets
+macro_rules! map {
+    // Single instance, with optional attribute
+    (
+        $( #[$cfg:meta] )?
+        $Sercom:ident, $IoSet:ident, $PadNum:ident, $PinId:ident, $Cfg:ident
```

Comment:
Is `IoSet` just ignored here?

Reply:
Yes. If you were able to catch that, you must have been reading very closely. Thanks for the attention.

I needed to transfer the Map implementations from IOSET to PinId to support the v1 API, but I already had the entire table and macro written for IOSET. I got lazy and just copied it from the other file and updated the macro. I can clean this up a bit.

Reply:
Probably not worth too much effort; I just wanted to confirm that the omission is intentional, which I figured was probably the case.